src/mbgl/renderer/render_tree.hpp

```diff
@@ -61,6 +62,7 @@ class RenderTree {
     // Render items
     virtual RenderItems getLayerRenderItems() const = 0;
     virtual RenderItems getSourceRenderItems() const = 0;
+    virtual std::vector<std::shared_ptr<gfx::Drawable>> getDrawables() const = 0;
```

Comment:
Can you return a const reference here? That way callers can decide if they want to make a copy or not.